R/get_smr.R

```diff
@@ -1,96 +1,162 @@
+#' is returned as a numeric vector.
+#'
+#' @references \href{http://documentation.sas.com/doc/en/pgmsascdc/9.4_3.4/statug/statug_stdrate_details.htm}{The STDRATE Procedure}
 #'
 #' @examples
 #' \dontrun{
 #' count <- 1500
-#' popn <- c(150000, 55000, 12000)
-#' std_rt <- c(0.001, 0.003, 0.05)
+#' popn <- c(13358, 13029, 34378, 39649, 11671)
+#' std_r <- c(0.004, 0.003, 0.007, 0.011, 0.013)
 #'
-#' get_smr(count, popn, std_rt)
-#' get_smr(count, popn, std_rt, output_type = "ratio")
-#' get_smr(count, popn, std_rt, output_type = "counts")
-#' get_smr(count, popn, std_rt, percent = FALSE)
-#' get_smr(count, popn, std_rt, percent = TRUE)
+#' get_smr(count = count, popn = popn, std_r = std_r, output_type = "ratio")
+#' get_smr(count = count, popn = popn, std_r = std_r, output_type = "counts")
+#' get_smr(count = count, popn = popn, std_r = std_r, percent = TRUE)
+#' get_smr(
+#'   count = count, popn = popn, std_r = std_r,
+#'   dist = "normal", interval = 0.95
+#' )
+#' get_smr(
+#'   count = count, popn = popn, std_r = std_r,
+#'   percent = TRUE, dist = "normal", interval = 0.95
+#' )
 #' }
 #'
 #' @importFrom magrittr %>%
+#' @importFrom rlang .data
 #' @export
-get_smr <- function(count, popn, std_rt, output_type = "ratio", percent = FALSE){
+get_smr <- function(count, popn, std_r, measure = "rate",
+                    output_type = "ratio", percent = FALSE,
+                    dist = NULL, interval = NULL) {
 
-  # validate inputs
+  # check validity of inputs
 
   if (!is.numeric(count)) {
-    stop("`count` must be numeric")
+    stop("`count` must be a non-negative integer of length 1")
   } else if (length(count) != 1) {
-    stop("`count` must be a scalar")
+    stop("`count` must be a non-negative integer of length 1")
+  } else if (count %% 1 != 0 | count < 0) {
+    stop("`count` must be a non-negative integer of length 1")
   }
 
-  if (length(popn) != length(std_rt)) {
-    stop("lengths of `popn` and `std_rt` must match")
+  if (length(popn) != length(std_r)) {
+    stop("lengths of `popn` and `std_r` must be the same")
   }
 
   if (!is.numeric(popn)) {
-    stop("`popn` must be a numeric vector")
-  } else if (!is.numeric(std_rt)) {
-    stop("`std_rt` must be a numeric vector")
+    stop("`popn` must be a numeric vector of non-negative integers")
+  } else if (any(popn < 0, na.rm = TRUE)) {
+    stop("`popn` must be a numeric vector of non-negative integers")
   }
 
-  if (!(output_type == "ratio" | output_type == "counts")) {
-    stop('`output_type` must be either "ratio" or "counts"')
+  # TODO: check if rates/risk are valid separately when risk SMR is added
+  if (!is.numeric(std_r)) {
+    stop("`std_r` must be a numeric vector")
   }
 
-  if (!percent %in% c(TRUE, FALSE)) {
-    stop("`percent` must be one of TRUE or FALSE")
+  # TODO: relax check when risk SMR is added
+  if (measure != "rate") {
+    stop("function currently only supports rate-SMR")
```

Comment:
change error message to reflect the comment above. 

Suggest: "function currently only supports rates from the standard population"

Don't forget to update the tests!